### ISSUE TITLE
[bug] no-unknown-property false positive on `<style precedence={...} />`

### ISSUE THREAD

**Issue by sadan4** [Rel1cx/eslint-react#1789]
### Describe the bug

no-unknown-property reports a false positive with  `<style precedence={...} />`

https://react.dev/reference/react-dom/components/style#props

### Reproduction

```tsx
function PieChart({data, colors}) {
  const id = useId();
  const stylesheet = colors.map((color, index) =>
    `#${id} .color-${index}: \{ color: "${color}"; \}`
  ).join();
  return (
    <>
      <style href={"PieChart-" + JSON.stringify(colors)} precedence="medium">
        {stylesheet}
      </style>
      <svg id={id}>
        …
      </svg>
    </>
  );
}
```

### Expected behavior

no false positive

### Platform and versions

```Text
System:
    OS: Linux 7.0 cpe:/o:nixos:nixos:26.05 26.05 (Yarara)
    CPU: (20) x64 13th Gen Intel(R) Core(TM) i5-13600KF
    Memory: 34.85 GB / 62.62 GB
    Container: Yes
    Shell: 5.9 - /run/current-system/sw/bin/zsh
  Binaries:
    Node: 25.9.0 - /etc/profiles/per-user/meyer/bin/node
    npm: 11.12.1 - /etc/profiles/per-user/meyer/bin/npm
    pnpm: 10.28.2 - /etc/profiles/per-user/meyer/bin/pnpm
    bun: 1.3.13 - /etc/profiles/per-user/meyer/bin/bun
    Deno: 2.7.14 - /etc/profiles/per-user/meyer/bin/deno
  Browsers:
    Brave Browser: 148.1.90.121
    Chromium: 148.0.7778.96
    Firefox: 150.0.2
    Firefox Developer Edition: 150.0.2
  npmPackages:
    @eslint-react/eslint-plugin: 3.0.0-beta.57 => 3.0.0-beta.57
```

### Stack trace

```Text
N/A
```
